src/platform/tls_openssl.c

```diff
@@ -810,10 +811,25 @@ CxPlatTlsSecConfigCreate(
     }
 
     if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_ENABLE_OCSP ||
-        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_DEFER_CERTIFICATE_VALIDATION) {
+        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_DEFER_CERTIFICATE_VALIDATION ||
+        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_REQUIRE_CLIENT_AUTHENTICATION) {
         return QUIC_STATUS_NOT_SUPPORTED; // Not supported by this TLS implementation
     }
 
+    if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_REVOCATION_CHECK_END_CERT ||
+        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_REVOCATION_CHECK_CHAIN ||
+        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_REVOCATION_CHECK_CHAIN_EXCLUDE_ROOT ||
+        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_IGNORE_NO_REVOCATION_CHECK ||
+        CredConfig->Flags & QUIC_CREDENTIAL_FLAG_IGNORE_REVOCATION_OFFLINE) {
+#ifdef CX_PLATFORM_USES_TLS_BUILTIN_CERTIFICATE
+        return QUIC_STATUS_NOT_SUPPORTED; // Only available on windows
+#else
+        if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_USE_TLS_BUILTIN_CERTIFICATE_VALIDATION) {
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+#endif
```

Comment:
I believe the `#ifdef` part is actually unnecessary. @thadhouse please correct me if I'm wrong.
```suggestion
        if (CredConfig->Flags & QUIC_CREDENTIAL_FLAG_USE_TLS_BUILTIN_CERTIFICATE_VALIDATION) {
            return QUIC_STATUS_INVALID_PARAMETER;
        }
```

Reply:
Well it depends on if openssl certificate loading supports those flags at all. Or is it just openssl cert validation that supports those flags?

Reply:
Only CAPI certificate chain building consumes those flags; OpenSSL doesn't consume them (I haven't looked to see if OpenSSL has analogues to these flags which we could translate between).  The `#ifdef` provides a shortcut for implementations that will never support these flags (i.e. platforms without CAPI, so not-Windows).  The USE_TLS_BUILDING_CERTF_VALIDATION flag check is in case on Windows platforms, the application has requested OpenSSL certificate validation, not CAPI.  In that case, I think it's actually up to the app to validate the certificate; does OpenSSL on Windows have its own trusted roots?